integration/Jenkinsfiles/shared.groovy

```diff
@@ -91,4 +91,14 @@ def withDockerCompose(String name, String file, String command, Closure post) {
     }
 }
 
+
+/**
+ * grep in server log file throwing an exception if errors occurs
+ */
+def failOnServerError(String logfile, String regexp='^[0-9]{4}-[0-9]{2}-[0-9]{2}.*ERROR.*') {
```

Comment:
Maybe something like
```groovy
if (new File(logfile).filterLine { line -> line ==~ /^\d{4}-\d{2}-\d{2}.*ERROR.*/ } as String) {
    currentBuild.result = Result.UNSTABLE
}
```

Would be more Java-ish ?

Reply:
agreed and it does not fork also

Reply:
that approach is not practicable as the pipeline script is evaluated on the master where the file is not available : [Pipeline file manipulation](https://support.cloudbees.com/hc/en-us/articles/230922508-Pipeline-Files-manipulation) for reference.